pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -390,20 +391,32 @@ func (r *ReconcileClusterDisruption) getOSDFailureDomains(clusterInfo *cephclien
 	for _, deployment := range osdDeploymentList.Items {
 		labels := deployment.Spec.Template.ObjectMeta.GetLabels()
 		failureDomainName := labels[topologyLocationLabel]
+		osdHostName := labels[osdHostNameLabel]
 		if failureDomainName == "" {
 			return []string{}, []string{}, errors.Errorf("failed to get the topology location label %q in OSD deployment %q",
 				topologyLocationLabel, deployment.Name)
 		}
 
+		// Assume node drain if osd deployment ReadyReplicas count is 0 and osd node is not schedulable
 		if deployment.Status.ReadyReplicas < 1 {
-			if !drainingFailureDomains.Has(failureDomainName) {
-				drainingFailureDomains.Insert(failureDomainName)
+			isNodeDrained, err := isNodeDrained(r.client, osdHostName)
+			if err != nil {
+				return []string{}, []string{}, errors.Wrap(err, "failed to check if node is drained")
```

Comment:
`s/[]string{}/nil/`

Reply:
update the return to use nil (instead of []string{})  in case of error.